src/main/java/com/thealgorithms/maths/Means.java

```diff
@@ -0,0 +1,57 @@
+package com.thealgorithms.maths;
+
+import java.util.stream.StreamSupport;
+import org.apache.commons.collections4.IterableUtils;
+
+/**
+ * https://en.wikipedia.org/wiki/Mean
+ * <p>
+ * by: Punit Patel
+ */
+public final class Means {
+
+    private Means() {
+    }
+
+    /**
+     * @brief computes the [Arithmetic Mean](https://en.wikipedia.org/wiki/Arithmetic_mean) of the input
+     * @param numbers the input numbers
+     * @throws IllegalArgumentException empty input
+     * @return the arithmetic mean of the input numbers
+     */
+    public static Double arithmetic(final Iterable<Double> numbers) {
+        if (!numbers.iterator().hasNext()) {
+            throw new IllegalArgumentException("Emtpy list given for Mean computation.");
+        }
```

Comment:
This piece is repeated in all three methods. I would suggest to add some method like
```java
private static void checkIfNotEmpty(final Iterable<Double> numbers) {
    if (!numbers.iterator().hasNext()) {
        throw new IllegalArgumentException("Emtpy list given for Mean computation.");
    }
}
```
and call if in every _mean_ method.

Side question: do you have some idea how to _enforce_ such check in methods calculating different methods added in the future?